src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_fill_emitter.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_fill_emitter.hpp"
+#include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_adr.h"
+#include "emitters/utils.hpp"
+
+using namespace Xbyak_aarch64;
+
+namespace ov {
+namespace intel_cpu {
+namespace aarch64 {
+
+using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
+using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
+using ExpressionPtr = ov::snippets::lowered::ExpressionPtr;
+
+jit_fill_emitter::jit_fill_emitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr)
+    : jit_emitter(h, isa, ov::element::f32, emitter_in_out_map::vec_to_vec) {
+    const auto fill = ov::as_type_ptr<snippets::op::Fill>(expr->get_node());
+    OV_CPU_JIT_EMITTER_ASSERT(fill != nullptr, "Expects Fill expression");
+    OV_CPU_JIT_EMITTER_ASSERT(fill->get_element_type().size() == 4,
+                              "Supports only 4 Byte element types but gets ", fill->get_element_type());
+
+    offset = fill->get_offset();
+    fill_value = fill->get_fill_value();
+    if (!is_optimized())
+        push_arg_entry_of("value", fill_value, true);
+    prepare_table();
+}
+
+size_t jit_fill_emitter::get_aux_gprs_count() const {
+    // Optimized version (fill full vector by zero) doesn't need additional register
+    if (is_optimized())
+        return 0;
+
+    return 1;
+}
+
+void jit_fill_emitter::emit_impl(const std::vector<size_t>& in,
+                            const std::vector<size_t>& out) const {
+    if (host_isa_ == dnnl::impl::cpu::aarch64::asimd) {
+        emit_isa<dnnl::impl::cpu::aarch64::asimd>(in, out);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("Fill emitter doesn't support ", host_isa_);
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_fill_emitter::emit_isa(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+    if (is_full_reg())
+        fill_full<isa>(out);
+    else
+        fill_tail<isa>(in, out);
+}
+
+template <>
+void jit_fill_emitter::emit_isa<dnnl::impl::cpu::aarch64::asimd>(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+    if (is_full_reg())
+        fill_full<dnnl::impl::cpu::aarch64::asimd>(out);
+    else
+        fill_tail<dnnl::impl::cpu::aarch64::asimd>(in, out);
+}
```

Comment:
Before I thought that you have `asimd` specific implementations for `jit_fill_emitter` and `fill_full<dnnl::impl::cpu::aarch64::asimd>(out);` call in template method baffled me. But if we take a look at the other emitter implementations, we well see that just exception throwing in `emit_impl` and no one specialized definitions.
So I'm sorry for the latest comment: can we leave only `template <cpu_isa_t isa> void jit_fill_emitter::emit_isa` but with `fill_full<isa>(out)` call as it's done now and implemented in other emitters? I mean to remove fully L58-64

Reply:
Applied. Thanks Alexandra!